gradle/plugins/src/main/kotlin/com.hedera.gradlebuild.java.gradle.kts

```diff
@@ -0,0 +1,153 @@
+    id("com.github.spotbugs")
+    id("org.sonarqube")
+    id("com.hedera.gradlebuild.java-base")
+    id("com.hedera.gradlebuild.publish")
+}
+
+@Suppress("UnstableApiUsage")
+testing.suites {
+    named<JvmTestSuite>("test") {
+        useJUnitJupiter()
+    }
+    register<JvmTestSuite>("integrationTest") {
+        testType = TestSuiteType.INTEGRATION_TEST
+        targets.all { testTask { group = "build" } }
+    }
+}
+
+tasks.withType<Test>().configureEach {
+    // NOTE: Uncomment to enable trace logs in the SDK during tests
+    // jvmArgs("-Dorg.slf4j.simpleLogger.log.com.hedera.hashgraph=trace")
+
+    // this task will fail on the first failed test
+    failFast = true
+
+    // emit logs per passed or failed test
+    testLogging {
+        exceptionFormat = FULL
+        events("passed", "skipped", "failed", "standardOut", "standardError")
+    }
+
+    // propagate system environment to test runner
+    systemProperty("OPERATOR_ID", providers.gradleProperty("OPERATOR_ID").getOrElse(""))
+    systemProperty("OPERATOR_KEY", providers.gradleProperty("OPERATOR_KEY").getOrElse(""))
+    systemProperty("CONFIG_FILE", providers.gradleProperty("CONFIG_FILE").getOrElse(""))
+    systemProperty("HEDERA_NETWORK", providers.gradleProperty("HEDERA_NETWORK").getOrElse(""))
+}
+
+tasks.withType<Javadoc>().configureEach {
+    options {
+        this as StandardJavadocDocletOptions
+        encoding = "UTF-8"
+        addStringOption("Xdoclint:all,-missing", "-quiet")
+        addStringOption("Xwerror", "-quiet")
+    }
+}
+
+tasks.jacocoTestReport {
+    // make sure to use any/all test coverage data for the report and run all tests before this report is made
+    executionData.from(
+        tasks.test.map { it.extensions.getByType<JacocoTaskExtension>().destinationFile!! },
+        tasks.named("integrationTest").map { it.extensions.getByType<JacocoTaskExtension>().destinationFile!! }
+    )
+
+    // remove generated proto files from report
+    classDirectories.setFrom(sourceSets.main.get().output.asFileTree.matching {
+        exclude(
+            "**/proto/**",
+            "**/AccountAllowanceAdjustTransaction.*",
+            "**/HederaPreCheckStatusException.*",
+            "**/HederaReceiptStatusException.*"
+        )
+    })
+
+    // configure it so only xml is generated for the report
+    reports {
+        xml.required = true
+        html.required = true
+        csv.required = false
+    }
+}
+
+
+// https://github.com/google/protobuf-gradle-plugin
+protobuf {
+    protoc {
+        artifact = "com.google.protobuf:protoc:3.24.3"
+    }
+    plugins {
+        id("grpc") {
+            artifact = "io.grpc:protoc-gen-grpc-java:1.58.0"
+        }
+    }
+}
+tasks.generateProto {
+    plugins { plugins.register("grpc") }
+}
+
+tasks.jar {
+    exclude("**/*.proto")
+    includeEmptyDirs = false
+}
+
+sourceSets.all {
+    configurations[getTaskName("", "compileProtoPath")].extendsFrom(configurations["internal"])
+}
```

Comment:
The Gradle prtotobuf plugin has its own "classpath" for the protobuf compiler tool. There is one for each "sourceSet" (e.g for _protobufs_ in main and protobufs in _test_). They are named after the source set. E.g. "compileProtoPath" for "main" or "testCompileProtoPath" for "test".

In Gradle, all classpaths (or module paths) are represented by the Gradle concept `Configuration`. Here we grep the configurations the protobuf uses and say they should inherit everything from the "internal" `Configuration`. Which is the one that has all the _versions_ attached we define centrally in `sdk/build.gradle.kts`

Unfortunately this kind of "glue" is often needed because version management is _complicated_ (sometimes you want everything to be consistent, sometimes you want different versions on different claspaths etc). And there are different solutions and no one-fit-all for all projects. That why some things in this area do not "just work".